prow/cmd/branchprotector/protect.go

```diff
@@ -276,11 +279,29 @@ func (p *protector) UpdateBranch(orgName, repo string, branchName string, branch
 		logrus.Infof("%s/%s=%s: already unprotected", orgName, repo, branchName)
 		return nil
 	}
+
+	currentBP, err := p.client.GetBranchProtection(orgName, repo, branchName)
```

Comment:
It would be preferable if we could list all the branch protections settings for a repo instead of requesting them individually based on Prow's branch protector config. Listing would save API tokens and more importantly, allow us to identify protection settings not explicitly mentioned in Prow's config that should be pruned.

Could you please evaluate if [this API endpoint](https://developer.github.com/v3/repos/branches/#list-branches) (which is already called [here](https://github.com/kubernetes/test-infra/blob/87aa9f5b3a9f96526e460810f360a5fd5afa6e7d/prow/cmd/branchprotector/protect.go#L250)) or any other provides enough fields to work as a cheaper alternative?

Reply:
Thanks for pointing this out. From my tests, it seems the [Get branch list endpoint](https://developer.github.com/v3/repos/branches/#list-branches) and [Get a single branch endpoint](https://developer.github.com/v3/repos/branches/#get-branch) both return a partially filled protection object. 
I looked at all available API endpoints, and did not find any other options for listing all branch protections either (perhaps I can make an enhancement request to github support?)

Reply:
Yeah opening an enhancement request ticket with GitHub support sounds like a good idea.
In the meantime if we are going to use the per branch `GetBranchProtection` calls please add a comment explaining why we have to do that instead of listing the protection for multiple branches at once.

Reply:
I have updated the PR. I also sent a request to github support. Will let you know once I hear back from them.

Reply:
Received a response from github support, saying they have added my +1 for this to their internal feature list (perhaps the `test-infra` team may consider requesting the same feature assuming  enhancement requests carry different weights depending on support level to which the repository/team has access).